### ISSUE TITLE
`Warning: `=destroy`(dest.value) can raise an unlisted exception: Exception` in refc in 2.0 but not devel with `std/isolation`/`isolate`

### ISSUE THREAD

**Issue by tersec** [nim-lang/Nim#23777]
### Description

```nim
{.push raises: [].}
import std/isolation
let literal = ""
discard isolate(literal)
```

### Nim Version

Warning occurs in refc:
```
Nim Compiler Version 2.0.7 [Linux: amd64]
Compiled at 2024-07-01
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: e98c98b46c6677ed6d1c9041d97ef6c2d414b654
active boot switches: -d:release
```

Warning does not occur in refc:
```
Nim Compiler Version 2.1.1 [Linux: amd64]
Compiled at 2024-07-01
Copyright (c) 2006-2024 by Andreas Rumpf

git hash: 4202b606b19f032e2a34056eaa384e46b5f97496
active boot switches: -d:release
```

### Current Output

```text
t.nim(4, 16) template/generic instantiation of `isolate` from here
nim20/lib/std/isolation.nim(37, 14) template/generic instantiation of `=destroy` from here
nim20/lib/std/isolation.nim(27, 6) template/generic instantiation from here
nim20/lib/std/isolation.nim(29, 13) Warning: `=destroy`(dest.value) can raise an unlisted exception: Exception [Effect]
```

### Possible Solution

Backport fix from devel

**Comment by narimiran** [comment]
`git bisect` led me to discovery that this was already reported: https://github.com/nim-lang/Nim/issues/23129

And the fix is https://github.com/nim-lang/Nim/pull/23195, plus https://github.com/nim-lang/Nim/pull/23198

----

I tested your example with those two fixes locally, and now it doesn't show the warning. If/when CIs are green on my test branch, I'll backport those two commits to `version-2-0`.

**Comment by narimiran** [comment]
> And the fix is #23195, plus #23198

...and it also needed https://github.com/nim-lang/Nim/pull/22975.

All three are now backported and I consider this fixed.
